src/xpk/core/core.py

```diff
@@ -2104,8 +2104,12 @@ def get_main_container(args, system, docker_image, resource_type) -> str:
         ' is required but not installed. Aborting"; exit 24;};'
     )
     xpk_internal_commands += (
+        'set -x;'
         'WORKER_ID=$HOSTNAME;'
-        f'gsutil -m cp -r /tmp/xla_dump/ {args.debug_dump_gcs}/$WORKER_ID;'
+        f'gcloud storage cp -r /tmp/xla_dump/ {args.debug_dump_gcs}/$WORKER_ID;'
+        f'gcloud storage cp /tmp/*.dump {args.debug_dump_gcs}/$WORKER_ID;'
+        f'gcloud storage cp -r /tmp/tpu_logs {args.debug_dump_gcs}/$WORKER_ID;'
+        'sleep 120;'
```

Comment:
What is the sleep for?

Reply:
This is a workaround, not an ideal solution. The main host pod requires a significant amount of time to upload `/tmp/*.dump` which are approximately 10 GiB in size.
I did find the main host pod was terminated earlier before finishing upload. I just want to add a buffer time to each pod's lifecycle, ensuring the host pod has sufficient time to finish uploading.

Let me know if you have better solution.

Reply:
Hmm buffer times are scary to put into production code because you don't know if you have set the time for too long or too short!

It looks like `gcloud storage cp ..` a blocking call? Why would the main host terminate earlier is strange... do you mind looking into this more? Thank you Zhiyu!